### ISSUE TITLE
Support for activation regularization on recurrent networks

### ISSUE THREAD

**Issue by soheilb** [fchollet/keras#2745]
Currently many layers such as Dense have activity_regularizer as a param but this is lacking in recurrent layers. I believe it would be useful to have this for recurrent layers as well. This allows to use either available or customized activity regularizer on the output of the recurrent networks. Currently one can use ActivityRegularization Layer to add available l1 l2 regularizer to any layer but that cannot be directly used for arbitrary regularizer.

**Comment by soheilb** [comment]
It seems the activityRegularization layer is not working as expected even for available regularizer. Here is  the relevant code to demonstrate:

```python
from keras.models import Model
from keras.layers.core import Dense, ActivityRegularization
from keras.layers.embeddings import Embedding
from keras.layers.recurrent import LSTM
from keras.layers import Input

x = Input(shape=(10,), dtype='int32')
o = Embedding(input_dim=10000, output_dim=100, mask_zero=True)(x)
o = LSTM(50)(o)
o = ActivityRegularization(l1=1000)(o)
o = Dense(1, activation='sigmoid')(o)

model = Model(input=x, output=o)
model.compile(loss='binary_crossentropy', optimizer='sgd')

print model.regularizers
```
It prints []. Tested with Keras 1.0.3.

**Comment by thouis** [comment]
This looks to be fixed in Keras 1.1.0:

```
>>> ...
>>> print(model.regularizers)
[<keras.regularizers.ActivityRegularizer object at 0x7f75580b05c0>]
```
